frontend/test/pytest/test_callback.py

```diff
@@ -395,5 +395,46 @@ def f(x):
     assert np.allclose(np.sin(1.0 / 2.0), f(1.0 / 2.0))
 
 
+@pytest.mark.parametrize(
+    "arg",
+    [0.1, jnp.array(0.1), jnp.array([0.1]), jnp.array([0.1, 0.2]), jnp.array([[1, 2], [3, 4]])],
+)
+def test_accelerate_device(arg):
+    """Test with device parameter"""
+
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+        return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+        return identity(x)
+
+    assert np.allclose(qjitted_fn(arg), arg)
+
+
+@pytest.mark.parametrize(
+    "arg",
+    [0.1, jnp.array(0.1), jnp.array([0.1]), jnp.array([0.1, 0.2]), jnp.array([[1, 2], [3, 4]])],
+)
+def test_accelerate_no_device(arg):
+    """Test with no device parameter"""
```

Comment:
@erick-xanadu are you able to assert that the function is being jitted and run on the default jax device?

Reply:
Same: I am not sure we can programmatically do this 🤔

Reply:
I mean, I can change the code to verify that the function is jitted, but verify that it runs on a specific requires a bit more digging.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/805/commits/fca8410bf789b1e691d0f34bca086b1d792af9b8